### PR TITLE
Fix bugs light simulation

### PR DESCRIPTION
**Pull request by peter-madigan**
OK I think I've got most of the egregious bugs sorted. There's still some tuning to the triplet lifetime and stuff like that, but I think it is ready to go. After you merge this PR, we can do a PR into the DUNE larnd-sim repo.

Primarily the issues I found were:
 - timestamps were not being saved correctly between light and charge because the timestamp lookup was done based on the "unique" event id. The unique event ids can differ between the light and charge (e.g. an event with no charge signal, but has a light signal)
  - light truth information was not being grabbed correctly in the loop
  - the light signals were being simulated in terms of the number of photoelectrons within a time interval - this has been changed to the "photocurrent" at a point in time, which makes more sense for interpolating and resampling

### CODE REVIEW

**Comment by peter-madigan** [comment]
@soleti This has all of the changes for the simulation of light waveforms

File: larndsim/light_sim.py [hunk 1]
```diff
@@ -230,10 +233,16 @@ def gen_light_detector_noise(shape, light_det_noise):
...
     noise_spectrum = cp.zeros((shape[0], desired_freq.shape[0]))
     for idet in range(shape[0]):
-        noise_spectrum[idet] = cp.interp(desired_freq, noise_freq, light_det_noise[idet] * cp.diff(noise_freq).mean(), left=0, right=0) / (bin_size)
+        noise_spectrum[idet] = cp.interp(desired_freq, noise_freq, light_det_noise[idet], left=0, right=0)
+    noise_spectrum *= cp.sqrt(cp.diff(noise_freq, axis=-1).mean()/bin_size) * LIGHT_DIGIT_SAMPLE_SPACING / LIGHT_TICK_SIZE
+    
     
     if shape[0]:
-        noise = cp.fft.irfft(noise_spectrum * cp.exp(1j * cp.random.uniform(size=noise_spectrum.shape) * 2* cp.pi), axis=-1)
+        noise = noise_spectrum * cp.exp(2j * cp.pi * cp.random.uniform(size=noise_spectrum.shape))
```
Comment (soleti) [inline larndsim/light_sim.py:241]: Can you add a comment in the code explaining what you're doing here?

File: larndsim/light_sim.py [hunk 2]
```diff
@@ -61,7 +64,7 @@ def sum_light_signals(segments, segment_voxel, light_inc, lut, start_time, light
 
             # find tracks that contribute light to this time tick
             for itrk in range(segments.shape[0]):
-                if light_inc[itrk,idet]['n_photons_det'] > 0:
+                if (light_inc[itrk,idet]['n_photons_det'] > 0):
```
Comment (soleti) [inline larndsim/light_sim.py:67]: do we need the parentheses?

File: larndsim/consts/light.py [hunk 3]
```diff
@@ -36,9 +36,9 @@
 LIGHT_DET_NOISE_SAMPLE_SPACING = 0.01 # us
 
 #: Total detector light threshold [ADC]
-LIGHT_TRIG_THRESHOLD = -2000
+LIGHT_TRIG_THRESHOLD = -10000
 #: Light digitization window [microseconds]
-LIGHT_TRIG_WINDOW = (1,1.56) # us
+LIGHT_TRIG_WINDOW = (0.9,1.66) # us
```
Comment (soleti) [inline larndsim/consts/light.py:41]: These are fine here, but maybe we should add them to the YAML file?

**Pull request merged**